src/agreements/CollateralAgreement.sol

```diff
@@ -143,29 +152,67 @@ contract CollateralAgreementFramework is IAgreementFramework {
         return agreement[id].finalizations >= agreement[id].party.length;
     }
 
+    function _isPartOfAgreement(uint256 id, address account) internal view returns (bool) {
+        return (
+            agreement[id].party.length > 0
+            && agreement[id].party[agreement[id].position[account].id] == account
+        );
+    }
+
+    function _addPosition(uint256 agreementId, PositionParams memory params) internal {
+        uint256 partyId = agreement[agreementId].party.length;
+        agreement[agreementId].party.push(params.party);
+        agreement[agreementId].position[params.party] = Position(
+            partyId,
+            params.balance,
+            PositionStatus.Idle
+        );
+        agreement[agreementId].balance += params.balance;
+    }
+
     /* ====================================================================== */
     /*                              IArbitrable
     /* ====================================================================== */
 
-    function settleDispute(uint256 id, PositionParams[] calldata settlement) public {
-        uint256 positionsLength = settlement.length;
-
+    function settleDispute(uint256 id, PositionParams[] calldata settlement) external override {
         if (msg.sender != arbitrator) revert OnlyArbitrator();
-        if (settlement.length < agreement[id].party.length) revert MissingPositions();
+        if (!agreement[id].disputed) revert AgreementNotDisputed();
+
+        uint256 positionsLength = settlement.length;
+        uint256 settleBalance;
 
+        if (positionsLength < agreement[id].party.length) revert MissingPositions();
         for (uint256 i = 0; i < positionsLength; i++) {
             if (i < agreement[id].party.length && agreement[id].party[0] != settlement[0].party)
                 revert PositionsMustMatch();
-            agreement[id].position[settlement[i].party] = Position(
+            settleBalance += settlement[i].balance;
+            _updatePosition(
+                id,
                 i,
-                settlement[i].balance,
+                settlement[i],
                 PositionStatus.Finalized
             );
-            if (i >= agreement[id].party.length) {
-                agreement[id].party.push(settlement[i].party);
-            }
         }
+        if (settleBalance > agreement[id].balance) revert DisputeBalanceMustMatch();
 
+        agreement[id].balance = settleBalance;
         agreement[id].finalizations = positionsLength;
     }
+
+    function _updatePosition(
```

Comment:
What does _update_ position mean? Maybe add some comments explaining this function?